packages/api/src/platforms/vtex/resolvers/validateSession.ts

```diff
@@ -154,6 +155,9 @@ export const validateSession = async (
       : null,
     geoCoordinates,
     city,
+    postalCode: isRepresentative
```

Comment:
question: I believe we should also check if there is already a postalCode set, otherwise we'll always update the postalCode to use the one from the saved address. Is that right?
```suggestion
    postalCode: isRepresentative && !postalCode
```

Reply:
@lariciamota in this case we really need to always pass the `postalCode` value, like `geoCoordinates`, `city` and others

Reply:
I don't know if I understood what you said..
But we've opened a [thread](https://vtex.slack.com/archives/C03L3CRCDC4/p1751038829652879) in Slack to understand the order of priority of postal code so we can make the necessary changes. Testing we've seen that neither B2B nor B2C flows are following what we thought should be the order.

Reply:
Humm, now I get it. I hadn't read this context before 🤔

Reply:
I'm changing a little bit the [approach](https://github.com/vtex/faststore/pull/2760/commits/e28042c389023ad2a5379c49e0c8c1f9b4971611) to get this session postalCode.  
Still running some tests.